ansible_rulebook/rule_set_runner.py

```diff
@@ -430,10 +431,18 @@ async def _call_action(
                     var_root = action_args.pop("var_root")
                     _update_variables(variables_copy, var_root)
 
-                action_args = {
-                    k: substitute_variables(v, variables_copy)
-                    for k, v in action_args.items()
-                }
+                if action == "debug":
+                    action_args = {
+                        k: substitute_variables(
+                            v, mask_sensitive_variable_values(variables_copy)
```

Comment:
@zkayyali812 this would break the action since it needs the real data not the masked data.

Reply:
This is only used in `debug` cases